pynxtools/dataconverter/helpers.py

```diff
@@ -313,7 +313,7 @@ def is_valid_data_field(value, nxdl_type, path):
             raise ValueError(f"The value at {path} should be of Python type: {accepted_types}"
                              f", as defined in the NXDL as {nxdl_type}.") from exc
 
-    if nxdl_type == "NX_POSINT" and not is_positive_int(value):
+    if nxdl_type in ["NX_POSINT", "NX_UINT"] and not is_positive_int(value):
```

Comment:
Related to the comment above. I wouldn't check for a virtual UINT representation. But I'd rather recommend using NX_POSINT.